### PR TITLE
feat: wallet_getCallsStatus

### PR DESCRIPTION
**Pull request by chris13524**
# Description

Adds support for `wallet_getCallsStatus` to Wallet Service according to [the spec](https://www.notion.so/walletconnect/API-specs-for-Wallet-Services-0c4b70a397e2435691f9bb1f20bc652d?pvs=4#5c4f865af7b24308920b13bae94da5ef).

This adds a basic happy-path test case to test this method. It mocks the call ID input with a "normal" Safe execution which we already have code for. However, going forward we will have code that will be able to test the other 2 `wallet_*` endpoints and we will be able to update this test case to be more E2E.

Depends on https://github.com/reown-com/yttrium/pull/37

## How Has This Been Tested?

New test case

## Due Diligence

* [ ] Breaking change
* [ ] Requires a documentation update
* [ ] Requires a e2e/integration test update

### CODE REVIEW

File: src/providers/pimlico.rs [hunk 1]
```diff
@@ -31,28 +33,28 @@ impl BundlerOpsProvider for PimlicoProvider {
     async fn bundler_rpc_call(
         &self,
         chain_id: &str,
-        id: u64,
-        jsonrpc: &str,
+        id: Id,
+        jsonrpc: Arc<str>,
```
Comment (geekbrother) [inline src/providers/pimlico.rs:37]: Nice catch 👍

File: src/test_helpers.rs
Comment (geekbrother) [inline src/test_helpers.rs:1]: Can we move this to `/tests` directory?  Looks like it fits better there.🙏

**Review by geekbrother — APPROVED** [review]
Great work on tests 👍 Thanks.

File: src/handlers/wallet/get_calls_status.rs [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+    #[error("Internal error")]
+    InternalError(GetCallsStatusInternalError),
+}
+
+#[derive(Error, Debug)]
+pub enum GetCallsStatusInternalError {}
+
+impl IntoResponse for GetCallsStatusError {
+    fn into_response(self) -> Response {
+        #[allow(unreachable_patterns)] // TODO remove
```
Comment (tomiir) [inline src/handlers/wallet/get_calls_status.rs:77]: is this ok?

File: src/handlers/wallet/get_calls_status.rs [hunk 3]
```diff
@@ -0,0 +1,453 @@
...
+            chain_id,
+        },
+        false,
+    ));
+
+    let receipt = provider
+        .get_user_operation_receipt(request.0 .0.user_op_hash)
+        .await
+        .unwrap();
+    // TODO handle None as CallStatus::Pending
```
Comment (tomiir) [inline src/handlers/wallet/get_calls_status.rs:135]: I guess this is a leftover?

File: src/test_helpers.rs
Comment (chris13524) [reply src/test_helpers.rs:1]: No because some of these helpers are used by unit tests inside `src`, and `src` doesn't have access to `tests`

File: src/handlers/wallet/get_calls_status.rs (hunk 2 above)
Comment (chris13524) [reply src/handlers/wallet/get_calls_status.rs:77]: Yeah this triggers because there's only 1 error that exists right now, `InternalError`, so clippy is saying I don't need the catch-all `e => ...` case. But we'll need that later when we add more error variants

File: src/handlers/wallet/get_calls_status.rs (hunk 3 above)
Comment (chris13524) [reply src/handlers/wallet/get_calls_status.rs:135]: This is some implementation work that's still required for best error handling. In the case where you submitted a user op but it hasn't been confirmed yet we need to return an appropriate response instead of panicking. This test case is captured below in `// TODO test case:`

Didn't implement in this PR as it's an edge case and wasn't sure we'd encounter it in the short-term, but I captured in this issue: https://linear.app/reown/issue/CR-553/wallet-getcallsstatus-handle-pending-case

File: src/providers/pimlico.rs (hunk 1 above)
Comment (chris13524) [reply src/providers/pimlico.rs:37]: Yeah doesn't hurt to be a little more generic and lean into alloy types. These types were required for something in this PR anyway
